build.gradle.kts

```diff
@@ -1,231 +1,37 @@
-import com.google.protobuf.gradle.generateProtoTasks
-import com.google.protobuf.gradle.id
-import com.google.protobuf.gradle.ofSourceSet
-import com.google.protobuf.gradle.plugins
-import com.google.protobuf.gradle.protobuf
-import com.google.protobuf.gradle.protoc
-import org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication
-
 plugins {
-    `java-library`
-    `maven-publish`
-    id("com.diffplug.spotless") version "6.4.2"
-    id("org.jsonschema2dataclass") version "4.2.0"
-    id("com.google.protobuf") version "0.8.17"
-    id("net.researchgate.release") version "3.0.0"
+    id("buildlogic.root-build")
+    // It does not support participating in precompiled script plugins
+    id("com.github.vlsi.stage-vote-release") version "1.82"
+    // The Kotlin Gradle plugin was loaded multiple times in different subprojects, which is not supported and may break the build.
+    kotlin("jvm") version "1.5.31" apply false
 }
 
-repositories {
-    mavenCentral()
-}
+val buildVersion = "${findProperty("version")}${releaseParams.snapshotSuffix}"
 
-java {
-    sourceCompatibility = JavaVersion.VERSION_11
-    targetCompatibility = JavaVersion.VERSION_11
-    withJavadocJar()
-    withSourcesJar()
-}
+println("Building Sigstore Java $buildVersion")
 
-tasks.withType<Javadoc> {
-    (options as StandardJavadocDocletOptions).addBooleanOption("Xwerror", true)
-    (options as StandardJavadocDocletOptions).addStringOption("sourcepath", "src/main/java")
-    // intentionally ignore missing errors for now
-    (options as StandardJavadocDocletOptions).addBooleanOption("Xdoclint:all,-missing", true)
-}
+val isReleaseVersion = rootProject.releaseParams.release.get()
 
-// TODO: keep until these code gen plugins explicitly declare dependencies
-tasks.named("sourcesJar") {
-    dependsOn(":generateProto", "::generateJsonSchema2DataClass0")
-}
-
-tasks.withType<Test> {
-    testLogging {
-        events("passed", "skipped", "failed")
-    }
-    // be very verbose in CI
-    if (environment.containsKey("CI")) {
-        testLogging {
-            showStandardStreams = true
-            showExceptions = true
-            exceptionFormat = org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
-        }
+releaseParams {
+    tlp.set("sigstore-gradle")
```

Comment:
is this the right value? sigstore-grade? (or should it be sigstore-java)

this releaseParams section is tiny bit confusing, but I guess it's from your release plugin?

Reply:
Yeah, that is indeed from the release plugin.
I created the plugin for Apache projects (Apache JMeter, Apache Calcite), so there's notion of "top level project", "organization", "component", etc.
The more I use the plugin for non-ASF projects, the more I want to make `tlp`, etc stuff optional.

However, I hope it does not hurt much. Some of the fields might be optional, however, I just copy-pasted the values which work.